### PR TITLE
Fix propagating spent coins IDs to the mempool on successful generator bundle creation in SpendSim's farm_block

### PR DESCRIPTION
**Pull request by AmineKhaldi**

### CODE REVIEW

File: chia/clvm/spend_sim.py [hunk 1]
```diff
@@ -263,9 +264,10 @@ async def farm_block(
                     generator_bundle = bundle
                     return_additions = additions
                     return_removals = bundle.removals()
+                    spent_coins_ids = [r.name() for r in return_removals]
 
                     await self.coin_store._add_coin_records([self.new_coin_record(addition) for addition in additions])
-                    await self.coin_store._set_spent([r.name() for r in return_removals], uint32(self.block_height + 1))
+                    await self.coin_store._set_spent(spent_coins_ids, uint32(self.block_height + 1))
```
Comment (arvidn) [inline chia/clvm/spend_sim.py:270]: I've struggled with the fact that the simulator doesn't use our normal code paths for running the chain. This was a particularly painful part, having to reach into an internal function of the coin store.

**Review by arvidn — APPROVED** [review]

File: chia/clvm/spend_sim.py [hunk 2]
```diff
@@ -253,6 +253,7 @@ async def farm_block(
         generator_bundle: Optional[SpendBundle] = None
         return_additions: List[Coin] = []
         return_removals: List[Coin] = []
+        spent_coins_ids = None
```
Comment (arvidn) [inline chia/clvm/spend_sim.py:256]: this line seems unnecessary. Is there any scenario where you access this variable before assigning a value to it?

Comment (AmineKhaldi) [reply chia/clvm/spend_sim.py:256]: An example code path is when `peak` is `None`.

Comment (arvidn) [reply chia/clvm/spend_sim.py:256]: I see

**Pull request merged**